src/quacc/calculators/espresso/utils.py

```diff
@@ -221,20 +222,331 @@ def espresso_prepare_dir(outdir: str | Path, binary: str = "pw") -> dict[str, An
                 "flvec": "matdyn.modes",
                 "fleig": "matdyn.eig",
             }
         },
-        "q2r": {"input": {"flfrc": "q2r.fc"}},
-        "bands": {"bands": {"filband": "bands.out", "outdir": outdir}},
-        "fs": {"fermi": {"file_fs": "fermi_surface.bxsf", "outdir": outdir}},
+        "q2r": {"input": {"fildyn": "matdyn", "flfrc": "q2r.fc"}},
+        "bands": {
+            "bands": {"prefix": "pwscf", "filband": "bands.out", "outdir": outdir}
+        },
+        "fs": {
+            "fermi": {
+                "prefix": "pwscf",
+                "file_fs": "fermi_surface.bxsf",
+                "outdir": outdir,
+            }
+        },
+        "dvscf_q2r": {
+            "input": {
+                "prefix": "pwscf",
+                "fildyn": "matdyn",
+                "outdir": ".",
+                "wpot_dir": Remove,
+            }
+        },
+        "postahc": {"input": {"ahc_dir": "ahc_dir/", "flvec": "matdyn.modes"}},
     }
 
     return outkeys.get(binary, {})
+
+
+def prepare_copy_files2(
+    parameters: dict[str, Any], binary: str = "pw"
+) -> dict[SourceDirectory, Filenames]:
+    """
+    Function that prepares the copy files for the espresso calculation.
+
+    Parameters
+    ----------
+    parameters
+        The input data for the espresso calculation
+    binary
+        The binary to use for the espresso calculation
+
+    Returns
+    -------
+    dict
+        The files to copy for the espresso calculation
+    """
+
+    to_copy = []
+
+    pw_base = [
+        Path("pwscf.save", "charge-density.*"),
+        Path("pwscf.save", "data-file-schema.*"),
+        Path("pwscf.save", "paw.*"),
+    ]
+
+    input_data = parameters.get("input_data", {})
+
+    if binary == "pw":
+        control = input_data.get("control", {})
+        restart_mode = control.get("restart_mode", "from_scratch")
+
+        electrons = input_data.get("electrons", {})
+        startingpot = electrons.get("startingpot", "atomic")
+        startingwfc = electrons.get("startingwfc", "atomic+random")
+
+        calculation = control.get("calculation", "scf")
+
+        if restart_mode == "restart":
+            to_copy.extend(
+                [
+                    Path("pwscf.wfc*"),
+                    Path("pwscf.mix*"),
+                    Path("pwscf.restart_k*"),
+                    Path("pwscf.restart_scf*"),
+                ]
+            )
+
+        need_chg_dens = (
+            startingpot == "file"
+            or calculation in ["bands", "nscf"]
+            or restart_mode == "restart"
+        )
+
+        need_wfc = startingwfc == "file" or restart_mode == "restart"
+
+        if need_chg_dens:
+            to_copy.append(Path("pwscf.save", "charge-density.*"))
+
+        if need_wfc:
+            to_copy.append(Path("pwscf.save", "wfc*.*"))
+
+        to_copy.extend(
+            [Path("pwscf.save", "data-file-schema.*"), Path("pwscf.save", "paw.*")]
+        )
+
+    if binary == "ph":
```

Comment:
Seems like these all should be `if`/`elif` blocks rather than individual `if` statements.